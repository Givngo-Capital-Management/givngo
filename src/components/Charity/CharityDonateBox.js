import React from 'react';
import { Button, Grid } from 'semantic-ui-react';
import DonateModal from '../Stripe/DonateModal';
import styles from './CharityDonateBox.module.scss';

export default function CharityDonateBox(props) {
  const { charity } = props

  const donateButton = (
    <Button className="donate ui right floated" color="orange">
      Donate
    </Button>
  )

  return (
    <div className={styles.box}>
      <div className={styles.header}>
        <p>Donate now</p>
      </div>
      <div className={styles.content}>
        <Grid>
          <div className="equal width row">
            <div className="ten wide column">
              <h3 className="ui header">
                <div className="content">
                  {`Donate to ${charity.name}`}
                </div>
              </h3>
            </div>
            <div className="six wide column">
              <DonateModal charity={charity} button={donateButton}/>
            </div>
          </div>
        </Grid>

        <h3>{`Total raised: $${charity.total_raised || 0}`}</h3>
      </div>
    </div>
  );
}
