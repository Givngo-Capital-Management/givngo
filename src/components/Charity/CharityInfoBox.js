import React from 'react';
import { Image, Button } from 'semantic-ui-react';
import styles from './CharityInfoBox.module.scss';

export default function CharityInfoBox(props) {
  const { charity } = props
  return (
    <div className={styles.box}>
      <Image
        className={styles.image}
        src={charity.logo}
        circular
      />
      <div className={styles.info}>
        <div className="charity-name">{charity.name}</div>
        <span>80k followers</span>
      </div>
      <Button className={styles.follow} color="orange">
        Follow
      </Button>
      <div className={styles.description}>
        <p>{charity.description}</p>
      </div>
    </div>
  );
}
