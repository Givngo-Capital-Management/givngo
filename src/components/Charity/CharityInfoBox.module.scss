.box {
  display: grid;
  grid: auto auto / 58px auto max-content;
  align-items: center;
  grid-row-gap: 16px;

  .image {
    grid-row: 1 / 2;
    grid-column: 1 / 2;
  }

  .info {
    grid-row: 1 / 2;
    grid-column: 2 / 3;

    .name {
      font-weight: 600;
      cursor: pointer;
    }
    .date {
      font-size: 13px;
      color: #707070;
    }
  }

  .follow {
    grid-row: 1 / 2;
    grid-column: 3 / 4;
  }

  .description {
    max-width: 615px;
    grid-row: 2 / 3;
    grid-column: 2 / 3;

    button {
      margin: 8px 0;
    }
  }
}