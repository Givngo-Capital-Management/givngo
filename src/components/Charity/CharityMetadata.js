import React from 'react';
import { Button, Divider, Icon } from 'semantic-ui-react';
import styles from './CharityMetadata.module.scss';
import moment from 'moment';

export default function CharityMetadata(props) {
  const { charity } = props;
  const date = moment.utc(charity.video_date * 1000).format('MMMM DD, YYYY').toString()
  return (
    <div className={styles.metadata}>
      <h3>{charity.video_title}</h3>
      <div className={styles.stats}>
        <span>{date}</span>
        <div className={styles.actions}>
          <Button basic icon labelPosition="left">
            <Icon name="share" />
            Share
          </Button>
        </div>
      </div>
      <Divider />
    </div>
  );
}
