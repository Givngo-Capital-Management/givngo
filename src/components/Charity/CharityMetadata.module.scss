.metadata {
  color: #707070;
  font-size: 16px;

  h3 {
    font-weight: 400;
    color: #111111;
  }

  .stats {
    display: flex;
    justify-content: space-between;
    width: 100%;
    align-items: center;

    .actions {
      display: flex;
      align-items: center;
      & > *:not(:last-child) {
        margin-right: 8px;
      }
    }
  }
}