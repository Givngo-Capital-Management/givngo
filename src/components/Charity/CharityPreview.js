import React from 'react';
import { Button, Image, Icon } from 'semantic-ui-react';
import styles from './CharityPreview.module.scss';
import moment from 'moment'
import Link from 'next/link';
import DonateModal from '../Stripe/DonateModal';

export default function CharityPreview(props) {
  const { horizontal, charity } = props;

  const id = charity ? charity.id : ""
  const name = charity ? charity.name : ""
  const videoDate = charity ? charity.video_date : ""
  const videoTitle = charity ? charity.video_title : ""
  const videoPreview = charity ? charity.video_preview : ""

  const date = moment.utc(videoDate * 1000).format('MMMM DD, YYYY').toString()

  const donateButton = (
    <Button className="mini compact ui orange basic smaller">
      <Icon name="heart outline" />
      Donate
    </Button>
  )

  return (
    <div
      className={[styles.preview, horizontal ? styles.horizontal : null].join(
        ' ',
      )}
    >
      <Link href={`/charity/${id}`}>
        <a>
          <div className={styles.image}>
            <Image src={videoPreview} />
            <div className={styles.time}>
              <span>05:22</span>
            </div>
          </div>
        </a>
      </Link>

      <div className={styles.info}>
        <Link href={`/charity/${id}`}>
          <a>
            <div className={`${styles.bold} ${styles.lines}`}>{videoTitle}</div>
          </a>
        </Link>
        <div className={styles.metadata}>
        <div className={styles.title}>{name}</div>
          <div>
            <span>{date}</span>
            <p>Short description</p>
          </div>
        </div>
        <DonateModal charity={charity} button={donateButton}/>
        <a href="www.google.ca" className={styles.link}>
          <Icon name="star outline" />
          Follow
        </a>
        <a href="www.google.ca" className={styles.link}>
          <Icon name="share" />
          Share
        </a>
      </div>
      
    </div>
  );
}
