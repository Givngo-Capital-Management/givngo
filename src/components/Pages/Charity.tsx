import React from 'react';
import {Grid} from 'semantic-ui-react';

import AppLayout from '../AppLayout';
import Video from '../Charity/Video';
import RelatedCharities from '../Charity/RelatedCharities';
import CharityMetadata from '../Charity/CharityMetadata';
import CharityInfoBox from '../Charity/CharityInfoBox';
import CharityDonateBox from '../Charity/CharityDonateBox';

import 'firebase/auth';

const Charity = (props: any) => {
  return (
    <AppLayout>
      <Video id="374481280" title="breast friends" />
      <div style={{ paddingLeft: 20, paddingRight: 20 }}>
        <Grid columns={2}>
          <Grid.Column width={10}>
            <CharityMetadata />
            <CharityInfoBox />
          </Grid.Column>
          <Grid.Column width={6} floated="right">
            <div>
              <CharityDonateBox />
              <RelatedCharities />
            </div>
          </Grid.Column>
        </Grid>
      </div>
    </AppLayout>
  )
};

export default Charity;