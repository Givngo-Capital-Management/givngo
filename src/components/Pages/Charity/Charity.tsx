import React, { useState, useEffect } from 'react';
import {Grid} from 'semantic-ui-react';

import AppLayout from '../../AppLayout';
import Video from '../../Charity/Video';
import RelatedCharities from '../../Charity/RelatedCharities';
import CharityMetadata from '../../Charity/CharityMetadata';
import CharityInfoBox from '../../Charity/CharityInfoBox';
import CharityDonateBox from '../../Charity/CharityDonateBox';

import firebase from 'firebase/app';
import 'firebase/database';
import {Charity} from '../../../types';
import { useRouter } from 'next/router';

const CharityPage = (props: any) => {
  const [charity, setCharity] = useState<Charity>();
  const router = useRouter();

  const { id } = router.query;

  useEffect(() => {
    const listener = firebase.database()
      .ref(`charities/${id}`)
      .on('value', (snapshot) => {
        if (snapshot) {
          setCharity(snapshot.val());
        }
      });
    return listener(null);
  }, []);

  return (
    <AppLayout>
      <Video id={charity?.video_id} title={charity?.video_title} />
      <div style={{ paddingLeft: 20, paddingRight: 20 }}>
        <Grid columns={2}>
          <Grid.Column width={10}>
            <CharityMetadata charity={charity ? charity : {}} />
            <CharityInfoBox charity={charity ? charity : {}} />
          </Grid.Column>
          <Grid.Column width={6} floated="right">
            <div>
              <CharityDonateBox charity={charity ? charity : {}}/>
              <RelatedCharities />
            </div>
          </Grid.Column>
        </Grid>
      </div>
    </AppLayout>
  )
};

export default CharityPage;