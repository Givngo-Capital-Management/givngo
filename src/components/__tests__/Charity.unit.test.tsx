import React from 'react';
import { shallow } from 'enzyme';
import Charity from '../../../pages/charity';

jest.mock('firebase/app', () => {
  return {
    auth: jest.fn(() => ({
      currentUser: ""
    })),
    apps: [],
    initializeApp: jest.fn()
  }
});

describe('Charity', () => {
  test('renders', () => {
    const wrapper = shallow(<Charity AuthUserInfo={{}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
