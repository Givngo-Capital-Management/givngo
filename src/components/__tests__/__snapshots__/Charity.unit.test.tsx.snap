// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charity renders 1`] = `
<ContextProvider
  value={
    Object {
      "AuthUser": null,
      "token": undefined,
    }
  }
>
  <WithAuthUserInfo(undefined) />
</ContextProvider>
`;
